examples/heartrate_monitor_central.py

```diff
@@ -0,0 +1,109 @@
+from bluezero import adapter
+from bluezero import central
+
+HRM_SRV = '0000180D-0000-1000-8000-00805f9b34fb'
+HR_MSRMT_UUID =      '00002a37-0000-1000-8000-00805f9b34fb'
+BODY_SNSR_LOC_UUID = '00002a38-0000-1000-8000-00805f9b34fb'
+HR_CTRL_PT_UUID =    '00002a39-0000-1000-8000-00805f9b34fb'
+
+def scan_for_heartrate_monitors(adapter_address=None, hrm_address=None, timeout=5.0):
+    # If there are multiple adapters on your system, this will scan using all of them
+    # Unless an address is specified and filtered
+    for dongle in adapter.Adapter.available():
+        # Filter dongles by adapter_address if specified
+        if adapter_address and adapter_address.upper() != dongle.address():
+            continue
+        
+        # Actually listen to nearby advertisements for timeout seconds
+        dongle.nearby_discovery(timeout=timeout)
+
+        # Iterate through discovered devices
+        for dev in central.Central.available(dongle.address):
+            # Filter devices if we specified a HRM address
+            if hrm_address and hrm_address == dev.address:
+                yield dev
+
+            # Otherwise, return devices that advertised the HRM Service UUID
+            if HRM_SRV.lower() in dev.uuids:
+                yield dev
+
+def on_new_heart_rate_measurement(iface, changed_props, invalidated_props):
+    value = changed_props.get('Value', None)
+    if not value:
+        return
+    
+    # Check bits field
+    if value[0] & 0x01:
+        # 16 bit heartrate value
+        hr = value[1] << 8 | value[2]
+    else:
+        # 8 bit heartrate value
+        hr = value[1]
```

Comment:
This library is aimed at students doing STEM. I am worried that this code is efficient but may be not that readable.
May be have a method called `is_hr_fmt_uint8`.
Elsewhere you have used `struct` library rather than bitwise operations. Would be good to be consistent so maybe:
```
if is_hr_fmt_uint8():
    hr, = struct.unpack_from("<B", value, 1)
else:
    hr, = struct.unpack_from("<H", value, 1)
````

Reply:
It's a little more complex now, but I switched to showing some good examples of struct since it's such a critical library for BLE